client/Routes.js

```diff
@@ -0,0 +1,98 @@
+
+import React, { Component } from 'react'
+import {connect} from 'react-redux'
+import { withRouter, Route, Switch, Redirect } from 'react-router-dom'
+import { Login, Signup } from './components/AuthForm';
+import Home from './components/Home';
+import { me, fetchCart, fetchProducts, fetchUsers } from './store'
+import Cart from './components/Cart';
+import Products from './components/Products';
+import Product from'./components/Product';
+import About from './components/About';
+import Orders from './components/Orders';
+import Order from './components/Order';
+
+/**
+ * COMPONENT
+ */
+class Routes extends Component {
+  componentDidMount() {
+    this.props.loadInitialData();
+    this.props.loadProducts();
+    this.props.loadCart();    
+  }
+  componentDidUpdate(){
+    this.props.loadUsers();
+    const url = window.location.origin;
+    console.log(url);
+    window.socket = new WebSocket(url.replace('http', 'ws'));
```

Comment:
i know you didnt learn this yet but imo socketio is much simpler to use 
https://socket.io/